File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int major = 0;
        map<int, int> freq;
        for(int x: nums){
            freq[x]++;
            if(freq[x] > (nums.size()/2)) major = x;
            
        }
        return major;
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        freq = {}
        for n in nums:
            freq[n] = 0
        for n in nums:
            freq[n] += 1
            if freq[n] > len(nums)/2:
                return n
        return 0
        
        